src/core/atomic.d

```diff
@@ -113,28 +113,30 @@ TailShared!T atomicLoad(MemoryOrder ms = MemoryOrder.seq, T)(ref const shared T
  *  newval = The value to store.
  */
 void atomicStore(MemoryOrder ms = MemoryOrder.seq, T, V)(ref T val, V newval) pure nothrow @nogc @trusted
-    if (__traits(compiles, { val = newval; }) && !is(T == shared S, S) && !is(V == shared U, U))
+    if (!is(T == shared S, S) && !is(V == shared U, U))
```

Comment:
Couldn't this just be `if (!is(T == shared) && !is(V == shared))`?

Reply:
Oh yeah... I guess so. I didn't think of that ;)